mutiny-core/src/lib.rs

```diff
@@ -816,7 +816,10 @@ impl<S: MutinyStorage> MutinyWallet<S> {
                         .pay_invoice(inv.clone(), labels.clone())
                         .await;
                     match payment_result {
-                        Ok(r) => return Ok(r),
+                        Ok(r) => {
+                            self.storage.set_invoice_labels(inv.clone(), labels)?;
```

Comment:
If it happens inside of `pay_invoice` for `node_manager` then it should happen inside `pay_invoice` for `fedimint_client`, or vice versa.

Reply:
That's what I wanted but the fedimint client doesn't have an instance of storage.

Reply:
Okay then they can both happen outside for now, that's probably better anyways so they both behave the same.

Reply:
Good call, did that for invoice creation as well. Cleaned up code a lot